src/main/java/build/buildfarm/admin/aws/AwsAdmin.java

```diff
@@ -148,4 +150,19 @@ private long getHostUptimeInMinutes(Date launchTime) {
     Calendar cal = Calendar.getInstance(TimeZone.getTimeZone("UTC"));
     return (cal.getTime().getTime() - launchTime.getTime()) / 60000;
   }
+
+  @Override
+  public void disableHostProtection(String scaleGroupName, String hostId) {
+    SetInstanceProtectionRequest disableRequest =
+        new SetInstanceProtectionRequest()
+            .withInstanceIds(hostId)
+            .withAutoScalingGroupName(scaleGroupName)
+            .withProtectedFromScaleIn(false);
+    SetInstanceProtectionResult response = scale.setInstanceProtection(disableRequest);
+    logger.log(
+        Level.INFO,
+        String.format(
+            "Disable protection of host: %s in AutoScalingGroup: %s and get result: %s",
+            hostId, scaleGroupName, response.toString()));
```

Comment:
we could check the response and log the error state with Level.SEVERE so its easier to find in cloudwatch, but we can do that later if needed.

Reply:
In the current implementation, this message will be not available in the grpc response. Worker will just send back response that it received the gracefully shutdown request and it wont wait until it finishes all actions, because they may take more than a couple of minutes. The worker will make a separate grpc call to AdminService that it's ready for losing protection. 

But, yeah, i just talked about this with Yuriy, we can make the caller wait for the response that including whether the worker is ready for shutdown, but that means AdminService is more coupled with workers. So we are not super sure yet.